libass/ass_fontconfig.c

```diff
@@ -88,14 +88,25 @@ static void destroy(void *priv)
     free(fc);
 }
 
-static void scan_fonts(FcConfig *config, ASS_FontProvider *provider)
+static void scan_fonts(FcConfig *config, ASS_FontProvider *provider, ASS_Library *lib)
 {
     int i;
     FcFontSet *fonts;
     ASS_FontProviderMetaData meta = {0};
 
-    // get list of fonts
-    fonts = FcConfigGetFonts(config, FcSetSystem);
+    // get list of fonts;
+    // sorting by default pattern prefers regular variants
+    FcPattern *pat = FcPatternCreate();
+    FcDefaultSubstitute(pat);
+    FcResult res;
+    // trim=FcFalse returns all system fonts
+    fonts = FcFontSort(config, pat, FcFalse, NULL, &res);
+    FcPatternDestroy(pat);
+    if (res != FcResultMatch) {
+        // fallback to unsorted full list
+        ass_msg(lib, MSGL_WARN, "fontconfig: failed to sort list of system fonts!");
+        fonts = FcConfigGetFonts(config, FcSetSystem);
```

Comment:
Can this return `NULL`?

Reply:
again not documented, but upon checking the code, yes if and only if `config` is `NULL` and obtaining/creating the default config fails.

However there *are* somet other relevant bits in docs:

> FcConfigReference must be called before using this  and  then  FcConfigDestroy when the return value is no longer referenced.

which we didn't before and still do after, oops
(The function is already explicitly documented as not MT safe, but i believe pending threading patches already take care of this anyway)

Reply:
Event hough docs technically require it, i think we don't actually need it since we hold on to the config object anyway even without refs. But either way, given it too can fail (and it makes cleanup more complicated), using it as a fallback no longer seemed to provide much benefit to me, so i dropped it entirely now